### DISCUSSION TITLE
Get error when exporting to Brainsight

### DISCUSSION THREAD

**Discussion by akelkar26** [simnibs/simnibs#190] [2. General]
Hi,

Hope you are well.

I am trying to use TMS Optimization and export the optimized target to Brainsight.

Please find attached the Python script used.

I used the Exporting from SimNibs steps here: https://simnibs.github.io/simnibs/build/html/documentation/neuronavigation/brainsight.html

However, I get this error after the optimal position is output on the command line:

Best coil position
[[ 0.49950636 -0.79859706 -0.33576201 26.1124801 ]
[-0.86180075 -0.49756089 -0.09865408 12.54552 ]
[-0.08827719 0.33863829 -0.93676638 75.18543527]
[ 0. 0. 0. 1. ]]

`[ simnibs ]CRITICAL: Uncaught exception AssertionError: Expecting array with shape (4, 4, N instrument marker).`

I saw a similar post on the old forum but it wasn't answered: https://mailman.drcmr.dk/pipermail/simnibs-discuss/2022-December/000435.html

Could you please help?

Awaiting your reply.
Thank you
Best,
Apoorva

[tms_optimization_code.docx](https://github.com/simnibs/simnibs/files/12480349/tms_optimization_code.docx)

**Comment by axthi** [comment]
Hi,
can you please try: 
brainsight().write(np.squeeze(opt_mat), fn)

Best,
Axel

**Comment by akelkar26** [comment]
Thank you so much Axel. That worked.
